### PR TITLE
misc: Defer Image loading for ImagePanel

### PR DESCRIPTION
**Pull request by Doomsdayrs**
### Description of the Change

Speeds up UI render time.

### Testing

- [x] Opened Instances Tab
- [x] Opened Servers Tab
- [x] Observed Nil Card

All seem to load faster then I can notice the loading bar.

Anyone with a slower system? 

### Related Issues

#759

### CODE REVIEW

**Comment by Doomsdayrs** [comment]
@RyanTheAllmighty should I leave this PR as is, or should I swap it to use a `SwingWorker`?

File: src/main/java/com/atlauncher/gui/components/ImagePanel.java [hunk 1]
```diff
@@ -22,31 +22,62 @@
...
-    public ImagePanel(Image image) {
-        this.image = image;
+    /**
+     * @param imageToLoad Deferred image loading
+     */
+    public ImagePanel(@Nonnull Callable<Image> imageToLoad) {
         this.setCursor(HAND);
-        this.setPreferredSize(new Dimension(Math.min(image.getWidth(null), 300), Math.min(image.getWidth(null), 150)));
+
+        add(new JLabel(Utils.getIconImage("/assets/image/loading-bars.gif")));
```
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/gui/components/ImagePanel.java:47]: I wouldn't show loading bars here, there's no network call, and loading from disk should be quick enough, the loading bars will look out of place

**Review by RyanTheAllmighty — CHANGES_REQUESTED** [review]

**Comment by RyanTheAllmighty** [comment]
> @RyanTheAllmighty should I leave this PR as is, or should I swap it to use a `SwingWorker`?

As long as it works, I don't mind :)

**Comment by Doomsdayrs** [comment]
> > @RyanTheAllmighty should I leave this PR as is, or should I swap it to use a `SwingWorker`?
> 
> As long as it works, I don't mind :)

Both work tbh.

SwingWorker seems like extra overhead, but integrates with Swing a bit better (is that even worth it?).

Thread is basic as can be on the other hand, with less integration.

**Pull request closed without merging**